modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.internal.field.business.type;
+
+import com.liferay.dynamic.data.mapping.form.field.type.constants.DDMFormFieldTypeConstants;
+import com.liferay.object.constants.ObjectFieldConstants;
+import com.liferay.object.constants.ObjectFieldSettingConstants;
+import com.liferay.object.field.business.type.ObjectFieldBusinessType;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.model.ObjectFieldSetting;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.util.SetUtil;
+import com.liferay.portal.vulcan.extension.PropertyDefinition;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@Component(
+	property = "object.field.business.type.key=" + ObjectFieldConstants.BUSINESS_TYPE_ENCRYPTED,
+	service = ObjectFieldBusinessType.class
+)
+public class EncryptedObjectFieldBusinessType
+	extends BaseObjectFieldBusinessType {
+
+	@Override
+	public Set<String> getAllowedObjectFieldSettingsNames() {
+		return SetUtil.fromArray(
+			ObjectFieldSettingConstants.NAME_MAX_LENGTH,
+			ObjectFieldSettingConstants.NAME_SHOW_COUNTER);
+	}
```

Comment:
@p-albuquerque do we need this settings?

Reply:
According to the mockup of the encrypted field edit sidebar, this field is able to support limit characters option, to this end, I set this settings support.
![image](https://user-images.githubusercontent.com/79872134/234863413-78d1bffc-add3-4713-84ed-a05b9d7bc757.png)

Reply:
got it, thank you @p-albuquerque